src/Statement/StatementListHolder.php

```diff
@@ -6,6 +6,7 @@
  * Interface for classes that contain a StatementList.
  *
  * @since 3.0
+ * @deprecated since 5.1
```

Comment:
I just realized this missed a migration path. What should I use instead? And that would be "use StatementListProvider instead", but the necessary guarantee that the provided list can be manipulated is missing. Adding this guarantee to the StatementListProvider is a breaking change. So in my opinion, adding this deprecation before 6.0 does not make much sense, because there is no alternative before 6.0.